Python/Product/ExternalProfilerDriver/VTuneToDWJSON.cs

```diff
@@ -209,6 +234,70 @@ class VTuneToDWJSON {
             return samples;
         }
 
+        /// <summary>
+        /// Creates a two-level dictionary from a stream of <cref>SampleWithTrace</cref>.
+        /// The "primary" (top-level) key is the module name, and the lower-level key
+        /// is the function name (function -> (sourcefile, base, size))
+        /// </summary>
+        public static Dictionary< string, Dictionary< string, FuncInfo > > ModuleFuncDictFromSamples(IEnumerable<SampleWithTrace> samples) {
+            var modFunDictionary = samples.SelectMany(sm => sm.AllSamples())
+                                              .Select(p => new { Module = p.Module, Function = p.Function, SourceFile = p.SourceFile })
+                                              .GroupBy(t => t.Module)
+                                              .Select(g => new { Module = g.Key,
+                                                                 Functions = g.Select(gg => new FuncInfo(gg.Function, gg.SourceFile)).Distinct(new FuncInfoComparer()),
+                                                               });
+
+            var mfdd = modFunDictionary.ToDictionary(r => r.Module,
+                                                     r => r.Functions.ToDictionary(
+                                                         rr => rr.FunctionName,
+                                                         rr => rr
+                                                     ));
+
+            return mfdd;
+        }
+
+        /// <summary>
+        /// Given a two-level module/function dictionary, check in <para>symbolPath</param> if
+        /// there is information on the source file/line number the function is defined in.
+        ///
+        /// TODO: maybe this function should take a list of possible directories to search?
+        /// </summary>
+        public static Dictionary<string, Dictionary<string, FunctionSourceLocation>> AddLineNumbers(ref Dictionary<string, Dictionary<string, BaseSizeTuple> > orig, string symbolPath)
```

Comment:
`ref` is only for when you want to change the value of the parameter, which isn't done here, ie there is no `orig = newvalue`